flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -286,20 +276,43 @@ public void setIdleStateRetentionTime(Time minTime, Time maxTime) {
 		}
 		minIdleStateRetentionTime = minTime.toMilliseconds();
 		maxIdleStateRetentionTime = maxTime.toMilliseconds();
+		setIdleStateRetentionDuration(Duration.ofMillis(minIdleStateRetentionTime));
+	}
+
+	/**
+	 * Specifies a retention time interval for how long idle state, i.e., state which
+	 * was not updated, will be retained.
+	 * State will never be cleared until it was idle for less than the retention time and will never
+	 * be kept if it was idle for more than the 1.5 * retention time.
+	 *
+	 * <p>When new data arrives for previously cleaned-up state, the new data will be handled as if it
+	 * was the first data. This can result in previous results being overwritten.
+	 *
+	 * <p>Set to 0 (zero) to never clean-up the state.
+	 *
+	 * <p>NOTE: Cleaning up state requires additional bookkeeping which becomes less expensive for
+	 * larger differences of minTime and maxTime. The difference between minTime and maxTime must be
+	 * at least 5 minutes.
+	 *
+	 * @param duration The retention time interval for which idle state is retained. Set to 0 (zero) to
+	 *                never clean-up the state.
+	 */
+	public void setIdleStateRetentionDuration(Duration duration){
+		configuration.set(ExecutionConfigOptions.IDLE_STATE_RETENTION, duration);
 	}
 
 	/**
 	 * @return The minimum time until state which was not updated will be retained.
 	 */
 	public long getMinIdleStateRetentionTime() {
```

Comment:
Add a new method `getIdleStateRetentionTime` and deprecate `getMinIdleStateRetentionTime` and `getMaxIdleStateRetentionTime`.